docs/ExpectAPI.md

```diff
@@ -79,6 +79,12 @@ declare global {
 }
 ```
 
+Depending on your TypeScript setup, you may also need this:
```

Comment:
It might be useful to add more details on the specific setup which requires this line. At the moment it looks puzzling. If `jest` is exported here, do you import it later to another module? Or perhaps simply an export is needed, because of `"isolatedModules": true` setting?

Reply:
> At the moment it looks puzzling

Absolutely.

Currently, you need `export = jest` when you create a new react app with the typescript template and the default configuration. IMO it's already an improvement to help people find a solution to this problem, even if it is "depending on your TypeScript setup".

`npx create-react-app my-app --template typescript`

```json
{
  "compilerOptions": {
    "target": "es5",
    "lib": [
      "dom",
      "dom.iterable",
      "esnext"
    ],
    "allowJs": true,
    "skipLibCheck": true,
    "esModuleInterop": true,
    "allowSyntheticDefaultImports": true,
    "strict": true,
    "forceConsistentCasingInFileNames": true,
    "noFallthroughCasesInSwitch": true,
    "module": "esnext",
    "moduleResolution": "node",
    "resolveJsonModule": true,
    "isolatedModules": true,
    "noEmit": true,
    "jsx": "react-jsx"
  },
  "include": [
    "src"
  ]
}
```

> It might be useful to add more details on the specific setup which requires this line. At the moment it looks puzzling. If jest is exported here, do you import it later to another module? Or perhaps simply an export is needed, because of "isolatedModules": true setting?

If you know _exactly_ when this is needed (and when not), it could be documented here. But I don't know about all the configurations which do / don't need this, so I don't feel confident in documenting it like that. Also, I don't know when this stopped working, assuming that it worked before...

Reply:
Hm.. Do you *import* the file with the declaration as current docs suggest: "declare the new matcher in the *imported* module". Or do you include the declaration in a `.d.ts` file? The thing is that declaration merging does not work in TypeScript without importing and `.d.ts` is not valid without `export` (an empty `export {}` does the job). Just guessing around.

Reply:
> Hm.. Do you _import_ the file with the declaration as current docs suggest: "declare the new matcher in the _imported_ module". Or do you include the declaration in a `.d.ts` file? The thing is that declaration merging does not work in TypeScript without importing and `.d.ts` is not valid without `export` (an empty `export {}` does the job) (it might be I am wrong here). Just guessing around.

Reply:
> it might be I am wrong here

No, you are right!

I have the type declarations in a separate `.d.ts` file, and your suggestion works:

`types/jest/index.d.ts`

```ts
declare global {
  namespace jest {
    interface Matchers<R> {
      toDoSomething(): R
    }
  }
}

export {}
```

I'll update the PR accordingly. Thanks! 🏅

Reply:
@mrazauskas I updated the commit, feel free to review it :)

Reply:
Done (;